components/engine_traits/src/engine.rs

```diff
@@ -86,6 +86,10 @@ pub trait KvEngine:
 #[derive(Debug, Clone)]
 pub struct SnapshotContext {
     pub range: Option<CacheRange>,
+    pub region_id: u64,
+    // the version of region epoch.
+    pub epoch_version: u64,
```

Comment:
But use `metapb::RegionEpoch` is not that convenient here, because we can't check the equality of `RegionEpoch ` since conf_version can be different. Use `RegionEpoch` but only check its `version` is somewhat weird to me.  I think the naming is enough to explain this field's meaning.